### DISCUSSION TITLE
How does calling R from Rust get packaged (or not)?

### DISCUSSION THREAD

**Discussion by sammysheep** [extendr/extendr#556] [Q&A] [answered]
How does calling R from Rust get packaged (or not)? 

Does it package some part of the runtime or does the macro expect a minimum version R installation to exist on the executing machine?

In my use case I have some simple plots written in R that'd I'd love to package without requiring a full R install. I am only using R base libraries to keep it really simple.

**Comment by Ilia-Kosenkov** [comment, marked as answer]
To call R from Rust you need to spin up R. There is no way of doing it without R installation. R library exposes a set of functions for starting and ending an R session, which can be conveniently accessed via `extendr-engine`. Once created, you can use all of the functionality of `extendr-api` to call into running R session, generate plots, etc, but it cannot be done without a local R installation.

**Reply by sammysheep** [reply]
That follows, thanks!
